src/hooks/useMembership.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+
+export const MEMBERSHIP_URL = process.env.MEMBERSHIP_URL;
+
+type Membership = {
+  status: 'pending' | 'active';
+};
+
+type FetchingStatus = boolean;
+
+const useMembership = (): [Membership | null, FetchingStatus] => {
+  const [membership, setMembership] = useState<Membership | null>(null);
+  const [fetching, setFetching] = useState(false);
+
+  useEffect(() => {
+    setFetching(true);
+    fetch(`${MEMBERSHIP_URL}/users/current.json`, {
+      // credentials: 'include',
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      }
+    }).then(async (res) => {
+      try {
+        const response = await res.json();
```

Comment:
I need to get the contract of the response and update here in order to consume it properly